### PR TITLE
fixing verify action for unfollow and follow actions.

### PR DESCRIPTION
**Pull request by sionking**
please refer to me talking we my self:
https://github.com/timgrossmann/InstaPy/issues/4103

This is a major bug in core functional.
** this has not been tested for **follow** please assist in testing.

**ANY CODE THAT LOOK FOR GENERAL TEXT IS UNSAFE AND NEED TO BE AVOIDED**

### CODE REVIEW

**Comment by sionking** [comment]
@timgrossmann is this place sleep ? We need more testers and devs.

**Comment by ghost** [comment]
Testing in my local.

**Comment by timgrossmann** [comment]
@sionking @halalfood Really sorry guys, but I'm quite short on time atm... :(

I'll try to test it, code looks good, though.
Thanks @halalfood for testing :blush:

**Comment by ghost** [comment]
unfollowing is working fine. waiting till my code gets to following then will report back in the morning.

File: instapy/unfollow_util.py [hunk 1]
```diff
@@ -1438,67 +1438,69 @@ def verify_action(browser, action, track, username, person, person_id, logger,
...
-                               "Back']"
+            post_action_text_correct = ["Follow", "Follow Back", "Unblock"]
+            post_action_text_fail = ["Following", "Requested"]
 
-        button_change = explicit_wait(browser, "VOEL",
-                                      [post_action_text, "XPath"], logger, 7,
-                                      False)
-        if not button_change:
-            reload_webpage(browser)
+        while not button_change:
```
Comment (sionking) [inline instapy/unfollow_util.py]: maybe a different logic can be set here... but this one works for now

**Comment by ghost** [comment]
I'm not noticing a difference from before and now so if code review passes, okay by me.

**Review by sionking — COMMENTED** [review]
> I'm not noticing a difference from before and now so if code review passes, okay by me.

I just changed the if condition. I think logic is the same. please verify if you can

**Comment by sionking** [comment]
I deeply checked follow and unfollow both works great on latest.

**Comment by MattB543** [comment]
> I deeply checked follow and unfollow both works great on latest.

Would this also fix the incorrect follow pool?

**Comment by sionking** [comment]
> > I deeply checked follow and unfollow both works great on latest.
> 
> Would this also fix the incorrect follow pool?

Show me an example/ thread.

**Comment by MattB543** [comment]
> > > I deeply checked follow and unfollow both works great on latest.
> > 
> > 
> > Would this also fix the incorrect follow pool?
> 
> Show me an example/ thread.

You said "Many users left followed but removed from follow pool."  

I think that's why my issue: [#4120](https://github.com/timgrossmann/InstaPy/issues/4120) is present.  

If I try to unfollow everyone that I've been following for more than 8 hrs (should be hundreds of people) it's like 50.  I think my follow pool has been corrupted and doesn't include hundreds of people that need to be unfollowed.

I guess also related to this issue? [#4110](https://github.com/timgrossmann/InstaPy/issues/4110)

**Review by timgrossmann — APPROVED** [review]
Code looking good 🙌 
Thank you very much for this fix! 🙏

**Comment by sionking** [comment]
> Code looking good 🙌
> Thank you very much for this fix! 🙏

I found some improvements need to be done

**Comment by timgrossmann** [comment]
@sionking Even better, could you create a new PR for it, please? 😊

**Pull request merged**